projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx

```diff
@@ -1,58 +1,139 @@
+                    const trimmedValue = item.value.trim();
                     return (
-                        <div class='dot-checkbox__container'>
+                        <div class={this.getClassName()}>
                             <input
                                 type='checkbox'
-                                name={item.value}
-                                checked={this.value === item.value ? true : null}
+                                disabled={this.shouldBeDisabled()}
+                                id={this.value}
+                                checked={this.value.indexOf(trimmedValue) >= 0 ? true : null}
                                 onInput={(event: Event) => this.setValue(event)}
-                                value={item.value}
+                                value={trimmedValue}
                             />
-                            <label htmlFor={item.value}>{item.label}</label>
+                            <label htmlFor={trimmedValue}>{item.label}</label>
                         </div>
                     );
                 })}
-                {this.hint ? <span class='dot-textfield__hint'>{this.hint}</span> : ''}
+                {this.hint ? <span class='dot-field__hint'>{this.hint}</span> : ''}
+                {!this.isValid() ? <span class='dot-field__error-message'>{this.requiredmessage}</span> : ''}
             </Fragment>
         );
     }
+
+    // Todo: find how to set proper TYPE in TS
+    private setValue(event): void {
+        this._dotPristine = false;
+        this._dotTouched = true;
+        this.value = this.getValueFromCheckInputs(event.target.value.trim(), event.target.checked);
+        this.emitValueChange();
+        this.emitStatusChange();
+    }
+
+    private getValueFromCheckInputs(value: string, checked: boolean): string {
+        const valueArray = this.value.trim().length > 0 ? this.value.split(',') : [];
+        const valuesSet = new Set(valueArray);
+        if (checked) {
+            valuesSet.add(value);
+        } else {
+            valuesSet.delete(value);
+        }
+        return Array.from(valuesSet).join(',');
+    }
+
+    private getClassName(): string {
+        return this.isValid() ? '' : 'dot-field__error';
+    }
+
+    private shouldBeDisabled(): boolean {
+        return this.disabled ? true : null;
+    }
+
+    private emitStatusChange(): void {
+        this.statusChange.emit({
+            name: this.name,
+            status: this.getStatus()
+        });
+    }
+
+    private getStatus(): DotFieldStatus {
+        return {
+            dotTouched: this._dotTouched,
+            dotValid: this.isValid(),
+            dotPristine: this._dotPristine
+        };
+    }
+
+    private isValid(): boolean {
+        return this.required ? !!this.value : true;
+    }
+
+    private emitValueChange(): void {
+        this.valueChange.emit({ name: this.name, value: this.value });
```

Comment:
this.value could be a String arrys then here we can translate it into a string join by ','

Reply:
that logic is being done here: https://github.com/dotCMS/core-web/blob/89121425ed5af19f0a0f962080be712abc59683f/projects/dotcms-field-elements/src/components/dot-checkbox/dot-checkbox.tsx#L90

because it's easier to handle an array of unique values there